vm/src/stdlib/binascii.rs

```diff
@@ -2,55 +2,13 @@ pub(crate) use decl::make_module;
 
 #[pymodule(name = "binascii")]
 mod decl {
-    use crate::byteslike::PyBytesLike;
+    use crate::byteslike::{PyAsciiBytesLike, PyBytesLike};
     use crate::function::OptionalArg;
-    use crate::obj::objbytearray::{PyByteArray, PyByteArrayRef};
-    use crate::obj::objbytes::{PyBytes, PyBytesRef};
-    use crate::obj::objstr::{PyString, PyStringRef};
-    use crate::pyobject::{PyObjectRef, PyResult, TryFromObject, TypeProtocol};
+    use crate::pyobject::PyResult;
     use crate::vm::VirtualMachine;
     use crc::{crc32, Hasher32};
     use itertools::Itertools;
 
-    enum SerializedData {
-        Bytes(PyBytesRef),
-        Buffer(PyByteArrayRef),
-        Ascii(PyStringRef),
-    }
-
-    impl TryFromObject for SerializedData {
-        fn try_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<Self> {
-            match_class!(match obj {
-                b @ PyBytes => Ok(SerializedData::Bytes(b)),
-                b @ PyByteArray => Ok(SerializedData::Buffer(b)),
-                a @ PyString => {
-                    if a.as_str().is_ascii() {
-                        Ok(SerializedData::Ascii(a))
-                    } else {
-                        Err(vm.new_value_error(
-                            "string argument should contain only ASCII characters".to_owned(),
-                        ))
-                    }
-                }
-                obj => Err(vm.new_type_error(format!(
-                    "argument should be bytes, buffer or ASCII string, not '{}'",
-                    obj.class().name,
-                ))),
-            })
-        }
-    }
```

Comment:
Maybe keep this? binascii does allow a bytearray, while struct doesn't:

```python
>>> binascii.a2b_hex(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: argument should be bytes, buffer or ASCII string, not 'int'
>>> struct.Struct(bytearray())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Struct() argument 1 must be a str or bytes object, not bytearray
```

Reply:
`PyAsciiBytesLike` aligns with what `binascii` accept more than this e.g. the following is allowed: `binascii.a2b_base64(array.array('B', b'1111'))`. I'm also not sure if `struct.Struct(bytearray())` should be disallowed or not. there is a comment about allowing it: https://github.com/python/cpython/blob/67acf74c4eaf64a860cc1bcda6efe6e9cb01f89b/Modules/_struct.c#L1466